src/app/utils/pinecone.ts

```diff
@@ -21,25 +20,34 @@ const getMatchesFromEmbeddings = async (embeddings: number[], topK: number, name
   // Retrieve the list of indexes
   const indexes = await pinecone.listIndexes()
 
+  let exists = false
```

Comment:
It won't let me comment on the line, but I would make line 18 above `const pinecone = new Pinecone();` and ditch the getter method entirely. More details in my other comment.

Reply:
Roger - done in 14650f5